packages/common/src/rule-engine/operators/helpers.ts

```diff
@@ -24,6 +37,16 @@ export abstract class BaseOperator<T = Primitive> {
 
   abstract evaluate(dataValue: Primitive, conditionValue: T): boolean;
 
+  extractValue(data: any, rule: Rule) {
```

Comment:
If there's no clear reason to use `any`, consider using `unknown` instead. It's safer to assume you don't know the type of the data in this case.